src/components/text-editor/TextEditor.jsx

```diff
@@ -1,31 +1,113 @@
+import "react-quill/dist/quill.snow.css";
 import "./TextEditor.css";
-import { BsPin } from "react-icons/bs";
+import { useState } from "react";
+import ReactQuill from "react-quill";
+import { useNotes } from "../../context/context";
 import { VscSymbolColor } from "react-icons/vsc";
 import { MdOutlineLabel } from "react-icons/md";
-import { AiOutlineDelete } from "react-icons/ai";
-import { BiArchiveIn } from "react-icons/bi";
+import { AiFillCloseCircle } from "react-icons/ai";
+import { Color, Label } from "../components";
+
+export function TextEditor({ showAddNote, setShowAddNote }) {
+  const [viewPopup, setViewPopup] = useState("");
+  const { noteState, noteDispatch, addNoteHandler, updateNoteHandler } =
+    useNotes();
+  const { content, color, tags, id } = noteState;
```

Comment:
you can destructure `notestate` in the above line 
Like : `{notestate : {content , color , tags , id}}`

Reply:
yeah, it was already too long in a line so did it in a new line.. thanks anyway!